tls/s2n_psk.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "crypto/s2n_tls13_keys.h"
+#include "tls/s2n_handshake.h"
+#include "tls/s2n_tls13_handshake.h"
+#include "tls/s2n_psk.h"
+#include "utils/s2n_safety.h"
+#include "utils/s2n_mem.h"
+
+int s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type)
+{
+    notnull_check(psk);
+
+    memset_check(psk, 0, sizeof(struct s2n_psk));
+    psk->hash_alg = S2N_HASH_SHA256;
```

Comment:
Should the hash algorithm always be SHA256? I thought this varied based on the psk?

Reply:
SHA256 is just a default:
> For externally established PSKs, the Hash algorithm MUST be set when the PSK is established or default to SHA-256 if no such algorithm is defined.

We can set the PSK to something different if a hash is provided.